backend/src/main/kotlin/org/darkSolace/muse/userModule/model/User.kt

```diff
@@ -0,0 +1,57 @@
+package org.darkSolace.muse.userModule.model
+
+import org.hibernate.Hibernate
+import org.hibernate.annotations.Cascade
+import org.hibernate.annotations.CascadeType
+import java.util.*
+import javax.persistence.*
+
+@Entity
+@Table(name = "museUser", uniqueConstraints = [UniqueConstraint(columnNames = ["username", "email"])])
+data class User(
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    var id: Long? = null,
+    val username: String,
+    var password: String,
+    var email: String,
+    var realName: String? = null,
+    @Temporal(TemporalType.TIMESTAMP)
+    val signUpDate: Date = Date(),
+    @Temporal(TemporalType.TIMESTAMP)
+    var lastLogInDate: Date? = null,
+    var bio: String? = null,
+    @Temporal(TemporalType.DATE)
+    var birthday: Date? = null,
+    var validatedAuthor: Boolean = false,
+    var onProbation: Boolean = false,
+    @OneToOne
+    @Cascade(CascadeType.ALL)
+    var userSettings: UserSettings = UserSettings(),
+    @ElementCollection
+    @Enumerated(EnumType.ORDINAL)
+    val userTags: MutableSet<UserTags> = mutableSetOf(),
```

Comment:
does this not need a '?' so that is may be null or is that implicit in case the set is null?

Reply:
in this current implementation `userTags` will never by `null`. On creation an empty set is assigned, containing zero entries (but not `null` entries). Over the users lifetime this set might be filled and emptied but the set itself always exists.